service/kvmclusterconfig/v1/resource/encryptionkey/delete_test.go

```diff
@@ -1 +1,69 @@
 package encryptionkey
+
+import (
+	"context"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_NewDeletePatch_Computes_Patch_Correctly(t *testing.T) {
+	testCases := []struct {
+		description    string
+		customObject   *v1alpha1.KVMClusterConfig
+		currentState   interface{}
+		desiredState   interface{}
+		expectedSecret *v1.Secret
+		expectedError  error
+	}{
+		{
+			description:    "delete existing secret when encryption key custom object is deleted",
+			customObject:   newCustomObject("cluster-1"),
+			currentState:   newEncryptionSecret(t, "cluster-1", map[string]string{}),
+			desiredState:   newEncryptionSecret(t, "cluster-1", map[string]string{}),
+			expectedSecret: newEncryptionSecret(t, "cluster-1", map[string]string{}),
+			expectedError:  nil,
+		},
+		{
+			description:    "nothing to delete when encryption key custom object get's deleted",
+			customObject:   newCustomObject("cluster-1"),
+			currentState:   nil,
+			desiredState:   newEncryptionSecret(t, "cluster-1", map[string]string{}),
+			expectedSecret: nil,
+			expectedError:  nil,
+		},
+		{
+			description:    "verify currentState type verification error handling",
+			customObject:   newCustomObject("cluster-1"),
+			currentState:   &v1.Pod{},
+			desiredState:   newEncryptionSecret(t, "cluster-1", map[string]string{}),
+			expectedSecret: nil,
+			expectedError:  wrongTypeError,
+		},
+	}
+
+	logger, err := micrologger.New(micrologger.DefaultConfig())
+	if err != nil {
+		t.Fatalf("micrologger.New() failed: %#v", err)
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			r, err := New(Config{
```

Comment:
Shouldn't this error be checked? Also, it looks that this instantiation could be moved out of the loop, it doesn't depend on `tc`.

Reply:
Added error check. I'll refactor resource instantiation out of loop in separate PR because it's a common pattern everywhere.